src/GitVersion.Core/VersionCalculation/SemanticVersioning/SemanticVersion.cs

```diff
@@ -159,29 +158,47 @@ public static bool TryParse(string version, string tagPrefixRegex, out SemanticV
 
             if (!parsed.Success)
             {
-                semanticVersion = null;
                 return false;
             }
 
             var semanticVersionBuildMetaData = SemanticVersionBuildMetaData.Parse(parsed.Groups["BuildMetaData"].Value);
             var fourthPart = parsed.Groups["FourthPart"];
             if (fourthPart.Success && semanticVersionBuildMetaData.CommitsSinceTag == null)
             {
-                semanticVersionBuildMetaData.CommitsSinceTag = int.Parse(fourthPart.Value);
+                semanticVersionBuildMetaData.CommitsSinceTag = Int32.Parse(fourthPart.Value);
+            }
+
+            int major = 0, minor = 0, patch = 0;
+
+            if (!parsed.Groups["Major"].Success || !Int32.TryParse(parsed.Groups["Major"].Value, out major))
+            {
+                return false;
+            }
+
+            if (parsed.Groups["Minor"].Success && !Int32.TryParse(parsed.Groups["Minor"].Value, out minor))
+            {
+                return false;
+            }
+            
+            if (parsed.Groups["Patch"].Success && !Int32.TryParse(parsed.Groups["Patch"].Value, out patch))
+            {
+                return false;
             }
 
             semanticVersion = new SemanticVersion
             {
-                Major = int.Parse(parsed.Groups["Major"].Value),
-                Minor = parsed.Groups["Minor"].Success ? int.Parse(parsed.Groups["Minor"].Value) : 0,
-                Patch = parsed.Groups["Patch"].Success ? int.Parse(parsed.Groups["Patch"].Value) : 0,
+                Major = major,
+                Minor = minor,
+                Patch = patch,
                 PreReleaseTag = SemanticVersionPreReleaseTag.Parse(parsed.Groups["Tag"].Value),
```

Comment:
We should probably change `SemanticVersionPreReleaseTag.Parse` to `SemanticVersionPreReleaseTag.TryParse` as well to have a consistent no-throwing semantics inside the `SemanticVersion.TryParse` method.

Reply:
Sure, will finish this up this evening. Noticed the last build failed, but not entirely sure why as couldn't see a failing test. Will debug this later.